src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/TransformRestTestCase.kt

```diff
@@ -89,5 +103,38 @@ abstract class TransformRestTestCase : IndexManagementRestTestCase() {
         assertEquals("Request failed", RestStatus.CREATED, response.restStatus())
     }
 
+    protected fun getTransform(
+        transformId: String,
+        header: BasicHeader = BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+    ): Transform {
+        val response = client().makeRequest("GET", "$TRANSFORM_BASE_URI/$transformId", null, header)
+        assertEquals("Unable to get transform $transformId", RestStatus.OK, response.restStatus())
+
+        val parser = newParser(response)
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser)
+
+        lateinit var id: String
+        var primaryTerm = SequenceNumbers.UNASSIGNED_PRIMARY_TERM
+        var seqNo = SequenceNumbers.UNASSIGNED_SEQ_NO
+        lateinit var transform: Transform
+
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            parser.nextToken()
+
+            when (parser.currentName()) {
+                _ID -> id = parser.text()
+                _SEQ_NO -> seqNo = parser.longValue()
+                _PRIMARY_TERM -> primaryTerm = parser.longValue()
+                Transform.TRANSFORM_TYPE -> transform = Transform.parse(parser, id, seqNo, primaryTerm)
+            }
+        }
+        return transform
+    }
+
     protected fun Transform.toHttpEntity(): HttpEntity = StringEntity(toJsonString(), APPLICATION_JSON)
+
+    protected fun newParser(response: Response): XContentParser {
```

Comment:
optional:

You can re-use the parser method in the ESTestCase this class is extending by just overriding the xContentRegistry method in ESTestCase to return the correct registry


```
// Where ever you are using newParsers(response) you could call instead
createParser(XContentType.JSON.xContent(), response.entity.content)

// and instead of newParser method you could have following override
override fun xContentRegistry(): NamedXContentRegistry {
        return NamedXContentRegistry(SearchModule(Settings.EMPTY, false, emptyList()).namedXContents)
    }

```